### PR TITLE
Update to newest full-moon

### PR DESCRIPTION
**Pull request by Kampfkarren**

### CODE REVIEW

**Comment by Kampfkarren** [comment]
!selenology

**Comment by Kampfkarren** [comment]
image

psh

**Review by matthargett — COMMENTED** [review]
unit test?

**Comment by Kampfkarren** [comment]
I need to push a commit again (renaming branches broke CI for a bit), will get to this when I can.

**Comment by Kampfkarren** [comment]
The newest full-moon, despite allocating a lot less stack space, STILL overflows, will post on Rust forums latre.

**Comment by Kampfkarren** [comment]
No amount of stacker or any nonsense like that seems to be fixing this bug...even when I disable the test that's overflowing we get a hang somewhere *else*. I'm dedicating time into this whenever I can, but sorry for all the delays!

**Comment by Kampfkarren** [comment]
Found it, or at least am getting closer--it's not a full-moon bug, `math.sin(1 + 1)` is hitting some recursion loop in standard library, probably because binary expressions were altered.

File: selene-lib/src/rules/standard_library.rs [hunk 1]
```diff
@@ -116,85 +120,85 @@ fn get_argument_type(expression: &ast::Expression) -> Option<PassedArgumentType>
...
-                    // Basic types will often re-implement these (e.g. Roblox's Vector3)
-                    ast::BinOp::Plus(_)
-                    | ast::BinOp::Minus(_)
-                    | ast::BinOp::Star(_)
-                    | ast::BinOp::Slash(_) => base,
+                // Basic types will often re-implement these (e.g. Roblox's Vector3)
+                ast::BinOp::Plus(_)
+                | ast::BinOp::Minus(_)
+                | ast::BinOp::Star(_)
+                | ast::BinOp::Slash(_) => get_argument_type(expression),
```
Comment (Kampfkarren) [inline selene-lib/src/rules/standard_library.rs:184]: 😛 There she is.

**Comment by Kampfkarren** [comment]
Ha! It caught a lint in zombie-strike that it didn't before, without any intentional changes.

**Pull request merged**